docs/launch_spec_experiments.py

```diff
@@ -0,0 +1,102 @@
+import os
+import sys
+from uuid import UUID
+
+from gem5art.artifact import Artifact
+from gem5art.run import gem5Run
+from gem5art.tasks.tasks import run_gem5_instance
+
+
+# Documentation
+gem5_repo = Artifact.registerArtifact(
+    command = 'git clone https://gem5.googlesource.com/public/gem5',
+    typ = 'git repo',
+    name = 'gem5',
+    path =  'gem5/',
+    cwd = './',
+    documentation = 'cloned gem5 master branch from googlesource on Nov 12nd'
+)
+
+gem5_binary = Artifact.registerArtifact(
+    command = 'scons build/X86/gem5.opt -j8',
+    typ = 'gem5 binary',
+    name = 'gem5',
+    cwd = 'gem5/',
+    path =  'gem5/build/X86/gem5.opt',
+    inputs = [gem5_repo,],
+    documentation = 'compiled gem5 binary right after downloading the source code'
+)
+
+m5_binary = Artifact.registerArtifact(
+    command = 'make -f Makefile.x86',
+    typ = 'binary',
+    name = 'm5',
+    path =  'gem5/util/m5/m5',
+    cwd = 'gem5/util/m5',
+    inputs = [gem5_repo,],
+    documentation = 'm5 utility'
+)
+
+packer = Artifact.registerArtifact(
+    command = '''wget https://releases.hashicorp.com/packer/1.4.5/packer_1.4.5_linux_amd64.zip;
+    unzip packer_1.4.5_linux_amd64.zip;
+    ''',
+    typ = 'binary',
+    name = 'packer',
+    path =  'disk-image/packer',
+    cwd = 'disk-image',
+    documentation = 'Program to build disk images. Downloaded sometime in November from hashicorp.'
+)
+
+disk_image = Artifact.registerArtifact(
+    command = './packer build spec2006/spec2006.json',
+    typ = 'disk image',
+    name = 'spec2006',
+    cwd = 'disk-image/',
+    path = 'disk-image/spec2006/spec2006-image/spec2006',
+    inputs = [packer, experiments_repo, m5_binary,],
+    documentation = 'Ubuntu Server with SPEC 2006 installed, m5 binary installed and root auto login'
+)
+
+linux_repo = Artifact.registerArtifact(
+    command = '''git clone https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/;
+    mv linux linux-stable''',
+    typ = 'git repo',
+    name = 'linux-stable',
+    path =  'linux-stable/',
+    cwd = './',
+    documentation = 'Linux kernel 4.19 source code repo obtained in November'
+)
+
+linux_binary = Artifact.registerArtifact(
+    name = 'vmlinux-4.19.83',
+    typ = 'kernel',
+    path = 'linux-stable/vmlinux-4.19.83',
+    cwd = 'linux-stable/',
+    command = '''git checkout v4.19.83;
+    cp ../config.4.19.83 .config;
+    make -j8;
+    cp vmlinux vmlinux-4.19.83;
+    ''',
+    inputs = [experiments_repo, linux_repo,],
+    documentation = "kernel binary for v4.19.83",
+)
+
+
+# Launch script
+if __name__ == "__main__":
+    cpu = ['kvm', 'atomic', 'o3']
+    benchmark = ['401.bzip2(base)', '403.gcc(base)', '464.h264ref(base)']
```

Comment:
Probably want to have all of the workloads here.

Reply:
I will do that after I finish debugging the script